packages/back-end/test/features.test.ts

```diff
@@ -0,0 +1,159 @@
+    const rawCondition = JSON.stringify({ id: "1234" });
+
+    expect(replaceSavedGroupsInCondition(rawCondition, groupMap)).toEqual(
+      JSON.stringify({ id: "1234" })
+    );
+  });
+
+  it("replaces the $inGroup and groupId with $in and the array of IDs", () => {
+    const ids = ["123", "345", "678", "910"];
+    const groupId = "grp_exl5jgrdl8bzy4x4";
+    groupMap.set(groupId, ids);
+
+    const rawCondition = JSON.stringify({ id: { $inGroup: groupId } });
+
+    expect(replaceSavedGroupsInCondition(rawCondition, groupMap)).toEqual(
+      '{"id":{"$in": ["123","345","678","910"]}}'
+    );
+  });
+
+  it("replaces the $notInGroup and groupId with $nin and the array of IDs", () => {
+    const ids = ["123", "345", "678", "910"];
+    const groupId = "grp_exl5jgrdl8bzy4x4";
+    groupMap.set(groupId, ids);
+
+    const rawCondition = JSON.stringify({ id: { $notInGroup: groupId } });
+
+    expect(replaceSavedGroupsInCondition(rawCondition, groupMap)).toEqual(
+      '{"id":{"$nin": ["123","345","678","910"]}}'
+    );
+  });
+
+  it("should replace the $in operator in and if the group.attributeKey is a number, the output array should be numbers", () => {
+    const ids = [1, 2, 3, 4];
+    const groupId = "grp_exl5jijgl8c3n0qt";
+    groupMap.set(groupId, ids);
+
+    const rawCondition = JSON.stringify({ number: { $inGroup: groupId } });
+
+    expect(replaceSavedGroupsInCondition(rawCondition, groupMap)).toEqual(
+      '{"number":{"$in": [1,2,3,4]}}'
+    );
+  });
+
+  it("should replace the $in operator in more complex conditions correctly", () => {
+    const ids = [1, 2, 3, 4];
+    const groupId = "grp_exl5jijgl8c3n0qt";
+    groupMap.set(groupId, ids);
+
+    const rawCondition = JSON.stringify({
+      number: { $inGroup: groupId },
+      id: "123",
+      browser: "chrome",
+    });
+
+    expect(replaceSavedGroupsInCondition(rawCondition, groupMap)).toEqual(
+      '{"number":{"$in": [1,2,3,4]},"id":"123","browser":"chrome"}'
+    );
+  });
+});
+
+it("should correctly replace the $in operator in advanced mode conditions", () => {
+  const ids = [1, 2, 3, 4];
+  const groupId = "grp_exl5jijgl8c3n0qt";
+  groupMap.set(groupId, ids);
+
+  const rawCondition = JSON.stringify({
+    $and: [
+      {
+        $or: [{ browser: "chrome" }, { deviceId: { $inGroup: groupId } }],
+      },
+      {
+        $not: [{ company: { $notInGroup: groupId } }],
+      },
+    ],
+  });
+
+  expect(replaceSavedGroupsInCondition(rawCondition, groupMap)).toEqual(
+    '{"$and":[{"$or":[{"browser":"chrome"},{"deviceId":{"$in": [1,2,3,4]}}]},{"$not":[{"company":{"$nin": [1,2,3,4]}}]}]}'
+  );
+});
+
+it("handle extra whitespace and spaces correctly", () => {
+  const ids = ["123", "345", "678", "910"];
+  const groupId = "grp_exl5jgrdl8bzy4x4";
+  groupMap.set(groupId, ids);
+
+  // eslint-disable-next-line
+  const rawCondition = JSON.stringify({ id: { $notInGroup: groupId } });
```

Comment:
In order to test whitespace, you can't use `JSON.stringify`.  You need to handcraft the strings yourself.  Something like this:

```js
{ "id": {"$inGroup"
  : 
              
                "grp_exl5jgrdl8bzy4x4"

}
}
```
should result in:
```js
{ "id": {"$in": ["123","345","678","910"]

}
}
```